vortex-ree/src/compute.rs

```diff
@@ -45,3 +52,39 @@ impl ScalarAtFn for REEArray {
         scalar_at(self.values(), self.find_physical_index(index)?)
     }
 }
+
+impl TakeFn for REEArray {
+    fn take(&self, indices: &dyn Array) -> VortexResult<ArrayRef> {
+        let primitive_indices = flatten_primitive(indices)?;
+        let physical_indices = match_each_integer_ptype!(primitive_indices.ptype(), |$P| {
+            primitive_indices
+                .typed_data::<$P>()
+                .iter()
+                .map(|idx| {
+                    self.find_physical_index(*idx as usize)
+                        .map(|loc| loc as u64)
+                })
+                .collect::<VortexResult<Vec<_>>>()?
+        });
+        take(self.values(), &PrimitiveArray::from(physical_indices))
```

Comment:
This is the same pattern that happens in Dict array, repeated value take on some dictionary.